### ISSUE TITLE
Service Worker not defined while accessing PWA in development

### ISSUE THREAD

**Issue by TeffenEllis** [coder/code-server#3834]
As first reported in a [PR](https://github.com/cdr/code-server/pull/3829) by @cuining, this regression seems to be introduced in the move away from Parcel. 

I believe this may be addressed in #3833, but some more information is still needed.

## OS/Web Information

- Web Browser: Chrome
- Local OS: macOS
- Remote OS: macOS
- Remote Architecture: 64 bit
- `code-server --version`: 3.11.0

## Steps to Reproduce

1. Start code-server via `yarn watch`
2. Open in Chrome, add a PWA shortcut
3. Refresh

## Expected

No console errors should occur.

## Actual

Service worker registration fails due to an invalid path. However, the file is available at an different path.

## Screenshot

<img width="1440" alt="Screen Shot 2021-07-21 at 14 40 39" src="https://user-images.githubusercontent.com/592134/126560278-33e099bd-9aa6-41b9-9d70-58e1f59ec6e0.png">

**Comment by jsjoeio** [comment]
Apologies @cuining for prematurely closing your PR and thank you for raising the issue! ♥️ (and @TeffenEllis for following up on this!)

## 3.11

I am not able to reproduce in 3.11, which is a good sign.

![image](https://user-images.githubusercontent.com/3806031/126561224-74cefd57-3f3d-486f-bc89-c31f58ddf893.png)

## Local development

I am also not able to reproduce 🤔 (macOS + Brave)

![image](https://user-images.githubusercontent.com/3806031/126562123-77a89d0d-197e-4275-8f3c-849888cd61ae.png)

**Comment by jsjoeio** [comment]
Wait, premature comment, I did not refresh

**Comment by jsjoeio** [comment]
## Video 


https://user-images.githubusercontent.com/3806031/126562428-49941d78-f023-4f31-8e65-99d822fed199.mov



Okay so actually I believe this is a different bug. I told @code-asher about this a while ago but couldn't reproduce consistently. This happened before the migration away from Parcel.

The workaround is to close the PWA and reopen. 

But the fact that you're seeing the raw JS in the browser feels odd and I wonder if these are two separate bugs or the same, manifesting differently 🤔

**Comment by lsoica** [comment]
I have the same issue with 3.11. It looks for the serviceWorker.js at /node_modules/code-server/dist/serviceWorker.js but in fact the new location is /node_modules/code-server/out/browser/serviceWorker.js

**Comment by jsjoeio** [comment]
Ahhh, I must have forgot to change it in one place. Thank you for the tip @lsoica!

**Comment by jsjoeio** [comment]
Could this be a caching issue? i.e. it cached the old path. Would a hard refresh fix it?

**Comment by lsoica** [comment]
Cleaning app cache and browser file cache didn't help. Anything else that I've missed ?

**Comment by jsjoeio** [comment]
Nope! Thank you for trying. cc @code-asher in case he has any other ideas. We're going to look at this in the next cycle (starting next week 🤞)

**Comment by code-asher** [comment]
Ack we didn't update this path!

https://github.com/cdr/code-server/blob/7b8cd25c0c48b6a1799587ebbc9b09d6af7988f3/src/browser/register.ts#L8

We probably didn't notice during development because we had the `dist` directory from previous builds.

**Comment by code-asher** [comment]
Looks like we should re-open that PR as it has the right fix.

**Comment by jsjoeio** [comment]
Cool! I'll do that since I closed it.

**Comment by Flcn** [comment]
Same for me but in 'production'. 
`[Service Worker] registration: Failed to register a ServiceWorker for scope ('https://DOMAIN/') with script ('https://DOMAIN/static/4e8cd09ef0412dfc7b148b7639a692e20e4fd6dd/usr/lib/code-server/dist/serviceWorker.js'): A bad HTTP response code (404) was received when fetching the script. TypeError: Failed to register a ServiceWorker for scope ('https://DOMAIN/') with script ('https:/DOMAIN/static/4e8cd09ef0412dfc7b148b7639a692e20e4fd6dd/usr/lib/code-server/dist/serviceWorker.js'): A bad HTTP response code (404) was received when fetching the script.`


The manifest look like this 

```
{
  "name": "code-server",
  "short_name": "code-server",
  "start_url": "./../../../../../../../..",
  "display": "fullscreen",
  "background-color": "#fff",
  "description": "Run editors on a remote server.",
  "icons": [
    {
      "src": "./../../../../../../../../static/4e8cd09ef0412dfc7b148b7639a692e20e4fd6dd/usr/lib/code-server/src/browser/media/pwa-icon-192.png",
      "type": "image/png",
      "sizes": "192x192"
    },
    {
      "src": "./../../../../../../../../static/4e8cd09ef0412dfc7b148b7639a692e20e4fd6dd/usr/lib/code-server/src/browser/media/pwa-icon-512.png",
      "type": "image/png",
      "sizes": "512x512"
    }
  ]
}

```

MacOS
3.11.1

**Comment by jsjoeio** [comment]
Thanks for the update! This _should_ be fixed when we get 3.11.2 out the door.
